core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/WorkflowActor.scala

```diff
@@ -28,6 +29,8 @@ abstract class WorkflowActor(val identifier: ActorVirtualIdentity)
   lazy val controlInputPort: ControlInputPort = wire[ControlInputPort]
   lazy val controlOutputPort: ControlOutputPort = wire[ControlOutputPort]
 
+  val promiseManager: PromiseManager
```

Comment:
Why not bring promiseManager in through wire?

Reply:
because workers and controller will have different handling logics.

Reply:
yes. And that is why I dont think handlers should be injected as dependencies. They should be added/registered. The promiseManager can be same for all types of actors.